export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+            bpy.ops.import_scene.gltf(filepath=input_gltf_filename)
+            obj = bpy.context.object
+            if obj.parent is not None:
+                continue
+            
+            object_name = obj.name.split('.')[0]
+            obj_dir = os.path.join(self.output_dir, object_name)
+            obj_info = actors_information[object_name]
+
+            joint_filename = input_gltf_filename.replace('.gltf', '_joints.json')
+            try:
+                with open(joint_filename, 'r') as f:
+                    joints_info = json.load(f)
+                joints_info = {j['name']: j for j in joints_info}
+            except FileNotFoundError:
+                joints_info = {}
+            
+            # traverse the tree, collect components info
+            geoms = []
+            components_info = {'static': {}, 'joints': {}}
+            # each element is (bpy object, directory_path, scale_factor, group_id)
+            # we apply the scale factor to the transforms here and save the resulting transforms in components.json.
+            # geometry scaling is already applied by UE when exporting geometries to GLTF.
+            q = deque([(obj, obj_dir, np.ones(3), object_name), ])
+            while len(q):
+                o, dir, scale_factor, group_id = q.popleft()
+                name = o.name.split('.')[0]
+                xyz, quat = get_relative_pose_and_center(o, scale_factor)
+
+                if o.children or ((o.type == 'MESH') and (o.data.name != 'SM_Dummy')):
+                    if name in components_info['static']:
+                        raise AssertionError(f'repeated static mesh {name}')
+                    components_info['static'][name] = {'pos': xyz, 'quat': quat}
+                    os.makedirs(dir, exist_ok=True)
+                    if o.children:  # StaticMeshComponent without any geometry (meant for grouping)
+                        for child in o.children:
+                            child_name = child.name.split('.')[0]
+                            if child_name in obj_info['geoms']:
+                                group_id = obj_info['geoms'][child_name]['decompose_group']
+                                group_id = f'{name}_{group_id}'
+                            else:
+                                group_id = None
+                            child_dir = osp.join(dir, child_name)
+                            q.append((child, child_dir, scale_factor*np.array(o.scale), group_id))
+                    else:  # StaticMeshComponent with geometry
+                        decompose_method = obj_info['geoms'][name]['decompose_method']
+                        geoms.append((dir, o, decompose_method, group_id))
+                elif (o.type == 'EMPTY') and not o.children:  # PhysicsConstraintComponent (i.e. joint)
+                    joint_name = utils.name_in_names(name, joints_info.keys())
+                    components_info['joints'][joint_name] = {
```

Comment:
Formatted inconsistently with line 329: `components_info['static'][name] = {'pos': xyz, 'quat': quat}`

Reply:
Do you mean that this dictionary should not be formatted multi-line? If that's what you mean, I have corrected it.